torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -2116,6 +2116,60 @@ AtenXlaType::native_batch_norm_backward(
                      : undefined);
 }
 
+std::tuple<at::Tensor, at::Tensor, at::Tensor> AtenXlaType::native_group_norm(
```

Comment:
I would have hoped for a real XLA lowering implementation 😉 
I would OK temporarily OK for something like this, but I'd move the core implementation of thes two functions into aten_tensor_ops.cpp/h and open an issue (no-stale) to implement this as real XLA lowering.

Reply:
@dlibenzi I feel for these composite ops, it should be fine to stay at `at::` level until we know lowering it might have some perf benefit. :P Otherwise it'll be pretty hard to keep up with PyTorch aten ops and a lot of them would be duplicated. ;) 
Btw what do you mean by `aten_tensor_ops.cpp/h`? Do you mean creating new files?

Reply:
> @dlibenzi I feel for these composite ops, it should be fine to stay at `at::` level until we know lowering it might have some perf benefit. :P Otherwise it'll be pretty hard to keep up with PyTorch aten ops and a lot of them would be duplicated. ;)
> Btw what do you mean by `aten_tensor_ops.cpp/h`? Do you mean creating new files?

So you are saying it might be OK to optimize C++ `virtual` function calls, but leaving that 100x cost there is OK? 😄 
That code, with all the extra cost of going through `at::` layer will get executed every time.
There is not graph hashing there which saves us.
If all the mambo-jumbo is instead behind a single IR node with dedicated raw XLA lowering, that will only get executed (but much cheaper since it is at raw XLA lowering level) only when hash changes (in which case it does not matter since it will be blended into a many-seconds XLA compilation times).

Yes, `aten_tensor_ops.cpp/h` need to be created, together with the github issue.

Reply:
Haha my point was they essentially generate the same HLO graph, although this way it has some `at::` dispatch overhead, they're usually hidden within  compilation time as well right? ;)

Reply:
> Haha my point was they essentially generate the same HLO graph, although this way it has some `at::` dispatch overhead, they're usually hidden within compilation time as well right? ;)

They might generate the same HLO, though some could be simplified.
They are not hidden behind compilation. These gets executed **every time**.
They might get hidden behind **execution time**, but I see no reason to stuff the clown car 😄